apps/lend/src/hooks/usePageOnMount.ts

```diff
@@ -32,26 +34,44 @@ function usePageOnMount(params: Params, location: Location, navigate: NavigateFu
   const walletSignerAddress = getWalletSignerAddress(wallet)
   const parsedParams = parseParams(params, chainIdNotRequired)
 
+  const handleReconnectWallet = useCallback(
+    async (walletState: Wallet, chainId?: number) => {
+      const label = walletState.label
+      await disconnect(walletState)
+      const newWalletState = (await connect({ autoSelect: { label, disableModals: true } }))[0]
+      if (chainId) await setChain({ chainId: ethers.toQuantity(chainId) })
+      return newWalletState
+    },
+    [connect, disconnect, setChain]
+  )
+
   const handleConnectCurveApi = useCallback(
     async (options: ConnectState['options']) => {
       if (options) {
         try {
           const [chainId, useWallet] = options
-          await updateProvider(wallet)
+          let parsedWallet = useWallet ? wallet : null
+
+          // check wallet sync again
+          if (useWallet && wallet && getWalletChainId(wallet) !== +chainId) {
```

Comment:
It would be good to add a little comment and provide a link to the onboard issue why this is being done.